### PR TITLE
return more helpful errors for link and unlink

### LINKED ISSUE #20: display error on `gofish install` to run `gofish link` when linking fails

### PR DESCRIPTION
**Pull request by tomlazar**
fixes #20

### CODE REVIEW

File: food.go [hunk 1]
```diff
@@ -104,7 +104,11 @@ func (f *Food) Install() error {
 	unarchiveOrCopy(cachedFilePath, barrelDir, u.Path)
 
 	// This is just a safety check to make sure that there's nothing there when we link the package.
-	f.Unlink(pkg)
+	err = f.Unlink(pkg)
+	if err != nil {
+		return fmt.Errorf("an error occurred while unlinking '%v' try running 'gofish unlink %v': %v", f.Name, err, f.Name)
```
Comment (bacongobbler) [inline food.go:109]: In general, it's best to keep CLI command suggestions out of the SDK. Separation of concerns and all.

Can you please return a concrete error type here like `UnlinkError`, and in `cmd/gofish/install.go`, can you check for that error type?

For example:

```
if err := food.Install(); err != nil {
    if _, ok := err.(*UnlinkError); ok {
        return fmt.Errorf("%s. Try running 'gofish unlink %s'", err.Error(), food.Name)
    }
    return err
}
```

https://blog.golang.org/go1.13-errors

File: food.go [hunk 2]
```diff
@@ -120,7 +124,7 @@ func (f *Food) Install() error {
 		}
 	}
 	if err := f.Link(pkg); err != nil {
-		return err
+		return fmt.Errorf("an error occurred while linking '%v' try running 'gofish link %v': %v", f.Name, err, f.Name)
```
Comment (bacongobbler) [inline food.go:127]: same here.

File: food.go (hunk 1 above)
Comment (bacongobbler) [reply food.go:109]: Please also use `%s` for string values.

Comment (tomlazar) [reply food.go:109]: was the error message good or do you want something different for the wording?

Comment (bacongobbler) [reply food.go:109]: the error message looks good.

**Pull request merged**